data-prepper-plugins/kafka-source/README.md

```diff
@@ -0,0 +1,131 @@
+# Kafka source
+
+This is the Data Prepper Kafka source plugin that reads records from Kafka topic. It uses the consumer API provided by Kafka to read messages from the broker.
+
+The Kafka source plugin supports OpenSearch 2.0.0 and greater.
+
+## Usages
+
+The Kafka source should be configured as part of Data Prepper pipeline yaml file.
+
+## Configuration Options
+
+```
+log-pipeline:
+  source:
+    kafka:
+      bootstrap_servers:
+        - 127.0.0.1:9092,localhost:9093, 127.0.0.1:9094
+      topics:
+        - topic:
+            name: my-topic-2
+            auth_type: SASL_OAUTH #optional
+            consumer:
```

Comment:
Please move `consumer` to set directly at the root level. It should be associated with the source, not with the topic.

```
log-pipeline:
  source:
    kafka:
      consumer:
        workers: 10
        autocommit: false
        ...
```